### PR TITLE
docs review and updates

### PR DESCRIPTION
**Pull request by vjd**
Through this PR, I am attempting to add all the missing information in the documentation especially with regards to the different methods of each function. There are so many dispatches for some of our exports that are not mentioned.

### CODE REVIEW

**Comment by pkofod** [comment]
The only thing I'm not 100% sure about here is if we want to advertise the `simobs(model, subject)` dispatches because we've discussed removing the init field. @andreasnoack

**Review by andreasnoack — COMMENTED** [review]
I agree with @pkofod that we probably don't want to advertise the `simobs(model, subject)` method.

**Comment by andreasnoack** [comment]
I've made some minor adjustments and then I have a suggestion.

File: docs/src/analysis/diagnostics.md [hunk 1]
```diff
@@ -1,17 +1,49 @@
...
+
+```julia
+my_inspect = DataFrame(inspect(my_fit))
+```
+
+The result of `inspect(my_fit)` is a `FittedPumasModelInspection` object, which can then
+be converted into tabular form using the `DataFrame` call. This is useful if one is interested
+to export the results into a spreadsheet, or use the resulting `DataFrame` for plotting.
+
+The prediction and residual diagnostics provided by the `FittedPumasModelInspection` can be summarized in the table below
```
[the remark this answers is not in the document]
Comment (vjd) [reply docs/src/analysis/diagnostics.md:25]: The diagnostics file is not complete yet. I plan to take this up in a different PR. Feel free to reject all the changes here.

I thought the table was useful :/ at least for me

[the remark this answers is not in the document]
Comment (andreasnoack) [reply docs/src/analysis/diagnostics.md:25]: We don't really expose most of these predictions and residuals so I don't think we should advertise them. I'll remove that part from this PR. Then let's discuss what to put in the diagnostics section in a separate PR.

**Comment by andreasnoack** [comment]
@vjd Are you okay with removing the `simobs(model, subject)` part as suggested by @pkofod?

**Comment by vjd** [comment]
yes, I think that's fine.

**Comment by vjd** [comment]
I removed the `simobs(model, subject)` as suggested. I think this is good to go? @andreasnoack

**Comment by andreasnoack** [comment]
I've made a few extra adjustments to that section. I'll merge once the tests pass.

**Pull request merged**